File: project-euler/001/solution.py
def gaussian_sum(n):
    return (n * (n + 1)) / 2

def solve():
    limit = 1000
    limit -= 1

    threes = 3 * gaussian_sum(limit // 3)
    fives = 5 * gaussian_sum(limit // 5)
    fifteens = 15 * gaussian_sum(limit // 15)

    return int(threes + fives - fifteens)

if __name__ == "__main__":
    print(solve())

File: project-euler/001/solution.cs
using System;

namespace Fifteen
{
	internal class Program
	{
		static int Gaussian_Sum(int n)
		{
			int result = n*(n+1) / 2;
			return result;
		}

		static void Main(string[] args)
		{
			int n = 1000;
			n--;

			int threes = Gaussian_Sum(n / 3) * 3;
			int fives = Gaussian_Sum(n / 5) * 5;
			int fifteens = Gaussian_Sum(n / 15) * 15;

			int result = threes + fives - fifteens;
			Console.WriteLine(result);
		}
	}
}
